helfomat-infrastructure-thwde/src/main/kotlin/de/helfenkannjeder/helfomat/infrastructure/thwde/ThwCrawlerOrganizationReader.kt

```diff
@@ -0,0 +1,237 @@
+package de.helfenkannjeder.helfomat.infrastructure.thwde
+
+import com.google.common.base.Preconditions
+import de.helfenkannjeder.helfomat.core.ProfileRegistry
+import de.helfenkannjeder.helfomat.core.geopoint.GeoPoint
+import de.helfenkannjeder.helfomat.core.organization.*
+import de.helfenkannjeder.helfomat.core.picture.DownloadFailedException
+import de.helfenkannjeder.helfomat.core.picture.PictureId
+import de.helfenkannjeder.helfomat.core.picture.PictureStorageService
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import org.jsoup.select.Elements
+import org.slf4j.LoggerFactory
+import org.springframework.batch.core.configuration.annotation.JobScope
+import org.springframework.batch.item.ItemReader
+import org.springframework.batch.item.ParseException
+import org.springframework.context.annotation.Profile
+import org.springframework.core.annotation.Order
+import org.springframework.core.io.ClassPathResource
+import org.springframework.stereotype.Component
+import org.springframework.util.StreamUtils
+import java.io.IOException
+import java.net.URL
+import java.nio.charset.Charset
+import java.util.*
+import java.util.regex.Pattern
+
+@Component
+@JobScope
+@Order(200)
+@Profile("!" + ProfileRegistry.DISABLE_THWDE_IMPORT)
+open class ThwCrawlerOrganizationReader constructor(
+    private val thwCrawlerConfiguration: ThwCrawlerConfiguration,
+    private val pictureStorageService: PictureStorageService
+) : ItemReader<Organization>, OrganizationReader {
+    private val logger = LoggerFactory.getLogger(ThwCrawlerOrganizationReader::class.java)
+    private val latitudePattern = Pattern.compile("lat = parseFloat\\((\\d+\\.\\d+)\\)")
+    private val longitudePattern = Pattern.compile("lng = parseFloat\\((\\d+\\.\\d+)\\)")
+
+    private var iterator: Iterator<Element>? = null
+    private var currentLetter = 'A'
+    private var currentPage = 1
+    private val logoPictureid = toPictureIdFromClasspathResource("thwde/logo.png")
+    private val teaserPictureId = toPictureIdFromClasspathResource("thwde/teaser.jpg")
+
+
+    override fun getName(): String {
+        return "thw-crawler"
+    }
+
+    @Throws(Exception::class)
+    override fun read(): Organization? {
+        if (iterator == null || !iterator!!.hasNext()) {
```

Comment:
no need for !!

Reply:
Extract a method with the content of this if that ensures to get a new iterator back